File: dashboard/src/types/incident.ts
export type IncidentStatus = 'New' | 'Acknowledged' | 'InProgress' | 'Resolved' | 'Closed';
export type IncidentSeverity = 'Critical' | 'High' | 'Medium' | 'Low' | 'Info';

export interface Incident {
    id: string;
    incidentId: string;
    title: string;
    severity: IncidentSeverity;
    status: IncidentStatus;
    category?: string;
    isAutoCreated: boolean;
    sourceType?: string;
    sourceName?: string;
    assignedTo?: UserSummary;
    slaResponseBreached: boolean;
    slaResolutionBreached: boolean;
    createdAt: string;
    acknowledgedAt?: string;
    resolvedAt?: string;
    duration?: string;
    customerId?: string;
    customerName?: string;
}

export interface IncidentDetail extends Incident {
    description?: string;
    sourceId?: string;
    assignedAt?: string;
    acknowledgedBy?: UserSummary;
    resolvedBy?: UserSummary;
    resolutionNotes?: string;
    rootCause?: string;
    closedBy?: UserSummary;
    closedAt?: string;
    sla: SlaInfo;
    impact?: string;
    affectedUsersCount?: number;
    firstOccurredAt?: string;
    lastOccurredAt?: string;
    occurrenceCount: number;
    resources: IncidentResource[];
    timeline: IncidentTimelineItem[];
    priorityScore: number;
}

export interface SlaInfo {
    responseDue?: string;
    resolutionDue?: string;
    responseBreached: boolean;
    resolutionBreached: boolean;
    timeToResponseDue?: string;
    timeToResolutionDue?: string;
}

export interface IncidentResource {
    id: string;
    resourceType: string;
    resourceId: string;
    resourceName: string;
}

export interface IncidentTimelineItem {
    id: string;
    eventType: string;
    title?: string;
    content?: string;
    oldValue?: string;
    newValue?: string;
    isInternal: boolean;
    isSystemGenerated: boolean;
    user?: UserSummary;
    createdAt: string;
}

export interface UserSummary {
    id: string;
    fullName: string;
    avatarUrl?: string;
}

export interface CreateIncidentRequest {
    title: string;
    description?: string;
    severity: IncidentSeverity;
    category?: string;
    customerId?: string;
    assignedToId?: string;
    impact?: string;
    affectedUsersCount?: number;
    resources?: CreateIncidentResourceRequest[];
}

export interface CreateIncidentResourceRequest {
    resourceType: string;
    resourceId: string;
    resourceName: string;
}

export interface AddCommentRequest {
    content: string;
    isInternal?: boolean;
}

export interface ResolveIncidentRequest {
    resolutionNotes?: string;
    rootCause?: string;
}

File: src/ERAMonitor.Core/Entities/Incident.cs
using ERAMonitor.Core.Enums;

namespace ERAMonitor.Core.Entities;

public class Incident : BaseEntity
{
    public Guid OrganizationId { get; set; }
    public Organization Organization { get; set; } = null!;
    
    public Guid? CustomerId { get; set; }
    public Customer? Customer { get; set; }
    
    public int IncidentNumber { get; set; } // Serial
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    
    public IncidentStatus Status { get; set; } = IncidentStatus.New;
    public IncidentSeverity Severity { get; set; } = IncidentSeverity.Medium;
    public IncidentPriority Priority { get; set; } = IncidentPriority.Medium;
    public string? Impact { get; set; }
    
    public string? SourceType { get; set; }
    public Guid? SourceId { get; set; }
    
    public Guid? AssignedToId { get; set; }
    public User? AssignedTo { get; set; }
    
    public Guid? AcknowledgedById { get; set; }
    public User? AcknowledgedBy { get; set; }
    
    public Guid? ResolvedById { get; set; }
    public User? ResolvedBy { get; set; }
    
    public DateTime? AcknowledgedAt { get; set; }
    public DateTime? ResolvedAt { get; set; }
    public DateTime? ClosedAt { get; set; }
    
    public int ResponseSlaMinutes { get; set; } = 15;
    public int ResolutionSlaMinutes { get; set; } = 240;
    public bool? ResponseSlaMet { get; set; }
    public bool? ResolutionSlaMet { get; set; }
    
    public string? RootCauseCategory { get; set; }
    public string? RootCauseDescription { get; set; }
    public string? ResolutionSteps { get; set; }
    public string? PreventiveActions { get; set; }
    
    public string? AffectedResources { get; set; } // JSONB
}
